Adds Rocky 8 repository config for local pulp

jovial opened the pull request:
We support Rocky 8 as a host OS with os_distribution == 'rocky'. This adds the configuration necessary to use the RPM snapshots from ark.

jovial committed:
Adds Rocky 8 repository config for local pulp

We support Rocky 8 as a host OS with os_distribution == 'rocky'. This
adds the configuration necessary to use the RPM snapshots from ark.

markgoddard reviewed (commented):
Looks like a good start. You'll need some changes in stackhpc.yml and dnf.yml to use the new repos.

jovial committed:
Abstract over version number v1

Unsure if it would be better to:

1) Update the remote on the repository
2) Change the publication to point to a new repository and keep
repositories for every minor version

Advantages of 1:

- Single repository in pulp

Advantages of 2:

- Less chance of conflicts (although I think an old repo is just an old snapshot)

jovial committed:
Fixups

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -164,6 +164,21 @@ stackhpc_pulp_distribution_deb_production:
 
+# Whether to sync Rocky Linux 8 packages.
+stackhpc_pulp_sync_rocky_8: "{{ os_distribution == 'rocky' }}"
+# Rocky 8 minor version number. Supported values: 6.
+stackhpc_pulp_sync_rocky_8_minor_version: 6
+# Rocky 8 Snapshot versions. The defaults use the appropriate version from
+# pulp-repo-versions.yml for the selected minor release.
+stackhpc_pulp_sync_rocky_8_appstream_version: "{{ lookup('vars', 'stackhpc_pulp_repo_rocky_8_%s_appstream_version' % stackhpc_pulp_sync_rocky_8_minor_version) }}"
+stackhpc_pulp_sync_rocky_8_baseos_version: "{{ lookup('vars', 'stackhpc_pulp_repo_rocky_8_%s_baseos_version' % stackhpc_pulp_sync_rocky_8_minor_version) }}"
+stackhpc_pulp_sync_rocky_8_extras_version: "{{ lookup('vars', 'stackhpc_pulp_repo_rocky_8_%s_extras_version' % stackhpc_pulp_sync_rocky_8_minor_version) }}"
+stackhpc_pulp_sync_rocky_8_nfv_version: "{{ lookup('vars', 'stackhpc_pulp_repo_rocky_8_%s_nfv_version' % stackhpc_pulp_sync_rocky_8_minor_version) }}"
+stackhpc_pulp_sync_rocky_8_powertools_version: "{{ lookup('vars', 'stackhpc_pulp_repo_rocky_8_%s_powertools_version' % stackhpc_pulp_sync_rocky_8_minor_version) }}"
```
minor nit: I'd have gone with `s/sync/repo/` here - the sync variables are flags

jovial replied:
Gone with your suggestion here, thanks.

jovial committed:
Address code review comment

Upadting variable name after feedback

jovial committed:
Merge branch 'stackhpc/xena' into feature/xena/rocky8-pulp

jovial committed:
Merge branch 'stackhpc/xena' into feature/xena/rocky8-pulp

jovial commented:
Sorry for all these updates. The merge button was greyed out. Doesn't explicitly say why, but I guess its because of "This branch is out-of-date with the base branch".

bbezak commented:
are you planning to add dnf.yml changes in this PR as well? @jovial

jovial commented:
> are you planning to add dnf.yml changes in this PR as well? @jovial

Yeah, I've done those here:

```
https://github.com/stackhpc/stackhpc-kayobe-config/pull/152
```
